### ISSUE TITLE
session is false why can not receive offline messages

### ISSUE THREAD

**Issue by dwin-wangjt** [eclipse/mosquitto#1081]

**Comment by ralight** [comment]
There are a few possibilities:

* You haven't used the same client id for all connections
* You haven't subscribed with QoS > 0
* No messages were published on the topics you subscribe to, and had QoS > 0
* You have the broker configured with `persistence false` (the default) and you restarted the broker

**Comment by alflexRH** [comment]
I am in the process of debugging a similar issue.
my publish command: `mosquitto_pub  -t 'test' -m 'test' -q 2`
my subscribe command: `mosquitto_sub -t 'test' -q 2 -c -i testing`

my sequence:
1) start subscriber
2) publish
-- message arrives
3) kill subscriber
4) publish again
5) start subscriber again
-- i now expect a message, but there is nothing. 

server is configured with 'persistence true'

**Comment by karlp** [comment]
works here.

```
karlp@strem:~$ mosquitto_sub -t 'test' -q 2 -c -i testing
^Z
[1]+  Stopped                 mosquitto_sub -t 'test' -q 2 -c -i testing
karlp@strem:~$ bg
[1]+ mosquitto_sub -t 'test' -q 2 -c -i testing &
karlp@strem:~$ 
karlp@strem:~$ 
karlp@strem:~$ mosquitto_pub -t 'test' -m 'test' -q 2
test
karlp@strem:~$ mosquitto_pub -t 'test' -m 'test' -q 2
test
karlp@strem:~$ fg
mosquitto_sub -t 'test' -q 2 -c -i testing
^C
(subscriber now offline)
karlp@strem:~$ mosquitto_pub -t 'test' -m 'test2' -q 2
karlp@strem:~$ mosquitto_pub -t 'test' -m 'test3' -q 2
karlp@strem:~$ mosquitto_pub -t 'test' -m 'test4' -q 2
( publish three more messages)
karlp@strem:~$ mosquitto_sub -t 'test' -q 2 -c -i testing &
[1] 26426
test2
test3
test4
karlp@strem:~$
( all messages received)
```

This is independent of persistence, that just determines whether stored messages persist across broker restarts

**Comment by alflexRH** [comment]
Thanks for testing. it convinced me it was a config-related error so i kept trying, and i found something: it happens when i set per_listener_settings to true.
this is my config:

per_listener_settings true
pid_file /var/run/mosquitto.pid
persistence true
persistence_location /var/lib/mosquitto/
listener 1883 127.0.0.1
protocol mqtt
max_connections 10
allow_anonymous true

If i set the per_listener_settings to false, it works as expected. may be there is another settings, apart from the listed ones: password_file, acl_file, psk_file, allow_anonymous, allow_zero_length_clientid, auth_plugin, auth_opt_*, auto_id_prefix, that is at play here?

**Comment by ralight** [comment]
Thanks for the follow up details, I can confirm that this was a bug and I've fixed it in my local branch will be part of 1.5.6 soon.
